File: app/src/main/java/com/tunjid/rcswitchcontrol/fragments/StartFragment.java
package com.tunjid.rcswitchcontrol.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.tunjid.rcswitchcontrol.R;
import com.tunjid.rcswitchcontrol.abstractclasses.BaseFragment;

public class StartFragment extends BaseFragment
        implements View.OnClickListener {

    public static StartFragment newInstance() {
        StartFragment startFragment = new StartFragment();
        Bundle args = new Bundle();
        startFragment.setArguments(args);
        return startFragment;
    }

    @Override @Nullable
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_start, container, false);
        View serverButton = rootView.findViewById(R.id.server);
        View clientButton = rootView.findViewById(R.id.client);

        serverButton.setOnClickListener(this);
        clientButton.setOnClickListener(this);

        return rootView;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.server:
                showFragment(InitializeFragment.newInstance());
                break;
            case R.id.client:
                showFragment(NsdScanFragment.newInstance());
                break;
        }
    }
}

File: app/src/main/java/com/tunjid/rcswitchcontrol/fragments/StartFragment.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Adetunji Dahunsi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.tunjid.rcswitchcontrol.fragments

import android.os.Bundle

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.tunjid.rcswitchcontrol.R
import com.tunjid.rcswitchcontrol.abstractclasses.BaseFragment
import com.tunjid.rcswitchcontrol.services.ServerNsdService

class StartFragment : BaseFragment(), View.OnClickListener {

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_start, container, false)
        val serverButton = rootView.findViewById<View>(R.id.server)
        val clientButton = rootView.findViewById<View>(R.id.client)

        serverButton.setOnClickListener(this)
        clientButton.setOnClickListener(this)

        return rootView
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.server -> {
                ServerNsdService.isServer = true
                showFragment(ClientNsdFragment.newInstance())
            }
            R.id.client -> showFragment(NsdScanFragment.newInstance())
        }
    }

    companion object {

        fun newInstance(): StartFragment {
            val startFragment = StartFragment()
            val args = Bundle()
            startFragment.arguments = args
            return startFragment
        }
    }
}
